src/lib/services/access-service.ts

```diff
@@ -118,27 +143,62 @@ export class AccessService {
         return this.store.getPermissionsForUser(user.id);
     }
 
-    getPermissions(): IPermission[] {
-        return this.permissions;
+    async getPermissions(): Promise<IAvailablePermissions> {
+        const bindablePermissions = await this.store.getAvailablePermissions();
+        const environments = await this.environmentStore.getAll();
+
+        const projectPermissions = bindablePermissions.filter((x) => {
+            return x.type === 'project';
+        });
+
+        const environmentPermissions = bindablePermissions.filter((perm) => {
+            return perm.type === 'environment';
+        });
+
+        const allEnvironmentPermissions = environments.map((env) => {
+            return {
+                name: env.name,
+                permissions: environmentPermissions.map((p) => {
```

Comment:
`p` is better, but if we need an abbreviation I would prefer `perm` - easier to associate with permission.

Reply:
Yep, previous comment taken on board - renamed to permission